### ISSUE TITLE
Doesn't detect files without extensions

### ISSUE THREAD

**Issue by thebyrd** [facebook/PathPicker#7]
You can reproduce this issue by putting a `Makefile` or `Dockerfile` in a directory and `ls | fpp`. It will work with other files in the directory that have an extension.

![](https://www.dropbox.com/s/p5kkxws8sa6231k/Screenshot%202015-05-07%2014.23.09.png?dl=1)

**Comment by sandyarmstrong** [comment]
Yeah, this is pretty annoying. I have the following little patch applied but it can cause some false positives when piping `git status` output, for example:

```
diff --git a/src/parse.py b/src/parse.py
index 8ee9175..1e2283e 100644
--- a/src/parse.py
+++ b/src/parse.py
@@ -22,10 +22,10 @@ import logger
 REPOS = ['www']

 MASTER_REGEX = re.compile(
-    '(\/?([a-z.A-Z0-9\-_]+\/)+[a-zA-Z0-9\-_.]+\.[a-zA-Z0-9]{1,10})[:-]{0,1}(\d+)?')
+    '(\/?([a-z.A-Z0-9\-_]+\/)+[a-zA-Z0-9\-_.]+)[:-]{0,1}(\d+)?')
 OTHER_BGS_RESULT_REGEX = re.compile(
     '(\/?([a-z.A-Z0-9\-_]+\/)+[a-zA-Z0-9_.]{3,})[:-]{0,1}(\d+)')
-JUST_FILE = re.compile('([a-zA-Z0-9\-_]+\.[a-zA-Z]{1,10})\s+')
+JUST_FILE = re.compile('([a-zA-Z0-9\-_.]+)\s+')
 FILE_NO_PERIODS = re.compile(
     '([a-zA-Z0-9\-_\/]{1,}\/[a-zA-Z0-9\-_]{1,})(\s|$|:)+')
```

Between this bug and https://github.com/facebook/PathPicker/issues/18, the released `fpp` is unusable on most of my repositories. But that diff makes it work well enough.

**Comment by mrjbq7** [comment]
Why not just use something like ``os.path.isfile(pathname)`` to detect files?

**Comment by sandyarmstrong** [comment]
@mrjbq7 on Hacker News, the devs said it's because users at Facebook keep their code mounted on network drives, and that would be slow. But they are open to adding it as an option.

**Comment by mrjbq7** [comment]
That makes sense - however, half of my files seem not to have extensions and so this doesn't work very well.  I also, don't see the need to differentiate between files and directories, since opening directories could be a valid use-case, too.

**Comment by pedrox** [comment]
After #50, this issue is actualy about dot files and with no period in the CWD.
It seems that those are not detected by design since it's more likely that they are not paths in the input. For the CWD I don't believe checking path existence would be a problem with network drive. Let's patch that.

**Comment by joni2back** [comment]
It makes sense, could be very hard to parse filenames without dots, taking into account that could be combined with another texts :disappointed:

**Comment by pcottle** [comment]
Yeah lets move this into #83. We can likely put the filesystem logic as an experimental flag which should mitigate any performance issues we might encounter. I do think it's the right future of the project though, since then we can match on all sorts of things and eliminate a ton of false positives
